### PR TITLE
#RTDSDK-839, implementing tokenWillExpire event

### PR DESCRIPTION
**Pull request by ddanila**
and some sanity tests for it

### CODE REVIEW

File: lib/accessmanager.js [hunk 1]
```diff
@@ -159,16 +168,24 @@ AccessManager.prototype.updateToken = function updateToken(newToken) {
...
       self.emit('tokenExpired', self);
     }, self.expires - Date.now());
 
+    clearTimeout(self._tokenWillExpireTimer);
+    self._tokenWillExpireTimer = setTimeout(function tokenWillExpireTimer() {
+      self._tokenWillExpireTimer = null;
+      self.emit('tokenWillExpire', self);
+    }, self.expires - Date.now() - 3 * 60 * 1000); // 3 minutes before expiration
+    // Negative timeout for setTimeout is fine and means "almost now"
+
```
Comment (markandrus) [inline lib/accessmanager.js:188]: Will you please document the event at the end of the file, like the other events?

**Comment by markandrus** [comment]
@ddanila one comment about missing JSDoc, but otherwise change looks good to me.

File: lib/accessmanager.js (hunk 1 above)
Comment (ddanila) [reply lib/accessmanager.js:188]: Sure, added a JSDoc for it.

**Comment by markandrus** [comment]
LGTM

**Pull request merged**